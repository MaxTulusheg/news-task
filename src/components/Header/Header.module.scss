@import '~/src/styles';

.wrapper {
  width: 100%;
  position: sticky;
  top: 0;
  z-index: 1;
  background-color: $color-bg-primary;
  box-shadow: 2px 0 8px $color-gray-300;
}

.navigation {
  display: flex;
  flex-direction: row;
  align-items: center;
  gap: .5rem;
  padding: .2rem 1rem;
  margin: 0 auto;
  height: 40px;
  max-width: 100%;
  
  @include tablet {
    height: 50px;
    max-width: 696px;
  }

  @include desktop {
    max-width: 960px;
  }

  @include largeDesktop {
    max-width: 1120px;
  }
}

.link {
  display: block;
  height: 100%;

  &:hover {
    cursor: pointer;
  }
}

.logo {
  display: block;
  height: 100%;
  width: auto;
}
