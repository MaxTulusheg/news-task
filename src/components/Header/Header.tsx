import React, { FC } from 'react';
import cn from 'classnames';

import logo from 'assets/images/logo.svg';

import styles from './Header.module.scss';

type HeaderProps = {
  className?: string;
};

const Header: FC<HeaderProps> = ({ className }) => (
  <header className={cn(styles.wrapper, className)}>
    <nav className={styles.navigation}>
      <a href="/" title="News" className={styles.link}>
        <img src={logo} alt="News logo" className={styles.logo} />
      </a>
    </nav>
  </header>
);

export default Header;
