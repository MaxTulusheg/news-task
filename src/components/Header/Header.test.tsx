import React from 'react';
import { render } from '@testing-library/react';

import Header from './Header';

describe('Header component', () => {
  it('renders correct structure', () => {
    const { container } = render(<Header />);

    expect(container).toMatchSnapshot();
  });
});
