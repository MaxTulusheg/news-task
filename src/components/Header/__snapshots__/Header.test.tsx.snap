// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders correct structure 1`] = `
<div>
  <header
    class="wrapper"
  >
    <nav
      class="navigation"
    >
      <a
        class="link"
        href="/"
        title="News"
      >
        <img
          alt="News logo"
          class="logo"
          src="logo.svg"
        />
      </a>
    </nav>
  </header>
</div>
`;
